### ISSUE TITLE
I have a question about the variable name of source code

### ISSUE THREAD

**Issue by Tommy-gif-ai** [zju3dv/snake#157]
Hello.
Thank you so much for your working. I read your paper , so interesting and attractive.
At the now I am seeing source code.

when I saw source code, I want know the meaning of variables.

trainers/rcnn_snake.py
        self.net = net

        **self.act_crit** = net_utils.FocalLoss()
        **self.awh_crit** = net_utils.IndL1Loss1d('smooth_l1')
        **self.cp_crit** = net_utils.FocalLoss()
        **self.cp_wh_crit** = net_utils.IndL1Loss1d('smooth_l1')
        **self.ex_crit** = torch.nn.functional.smooth_l1_loss
        **self.py_crit** = torch.nn.functional.smooth_l1_loss

I read paper, I guess   ex : extreme point .
but the others I don't know....
Please answer to me~~

**Comment by pengsida** [comment]
Which variable?

**Comment by Tommy-gif-ai** [comment]
> Which variable?

Thank you for your reply.

1. act_crit
2. awh_crit
3. cp_crit
4. cp_wh_crit
5. ex_crit
6. py_crit

I want know the meaning of six variables.

And, Additionally  I have a quetion.

In the paper, the input of Deep snake is contour.

Is the data type of the contour image? Is it Point? I'm curious about the data type.

Thank you so much.

**Comment by pengsida** [comment]
https://github.com/zju3dv/snake/blob/master/lib/datasets/cityscapes/rcnn_snake.py#L235

**Comment by pengsida** [comment]
1. they are loss functions for learning detector, extreme points and polygon.
2. Contour points.

**Comment by Tommy-gif-ai** [comment]
> 1. they are loss functions for learning detector, extreme points and polygon.
> 2. Contour points.

Really Thank you very much~~
